### ISSUE TITLE
Bug: Cannot use built-in configs

### ISSUE THREAD

**Issue by PabloLION** [typescript-eslint/typescript-eslint#5031]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Issue Description

From issue #5030 
### Test

```bash
yarn install && yarn lint
```

### Problem

1. In the first commit on this branch, the `yarn lint` command works correctly.
2. In the second commit, the `yarn lint` command fails.

### Reproduction

1. Run `yarn init` to create a whatever new repo.
2. Install packages with `yarn` following [Getting Started - Linting your TypeScript Codebase](https://typescript-eslint.io/docs/linting/)

   ```bash
     yarn add --dev eslint typescript @typescript-eslint/parser @typescript-eslint/eslint-plugin
   ```

3. Add this `eslintConfig` to `package.json`, then run `yarn lint`. The command works.

   ```json
   "eslintConfig": {
       "env": {
         "node": true
       },
       "extends": [
         "eslint:recommended",
         "plugin:@typescript-eslint/recommended"
       ],
       "parser": "@typescript-eslint/parser",
       "plugins": [
         "@typescript-eslint"
       ],
       "root": true
     },
   ```

4. Add these two lines to `eslintConfig.extends`, then the command fails.

   ```json
   "plugin:@typescript-eslint/recommended-requiring-type-checking",
   "plugin:@typescript-eslint/strict"
   ```

### Env

I'm using MacBook Pro (2021, 16'), which has M1 chips.

VSCode

```plaintext
Version: 1.67.1
Commit: da15b6fd3ef856477bf6f4fb29ba1b7af717770d
Date: 2022-05-06T12:37:14.619Z
Electron: 17.4.1
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Darwin arm64 21.3.0
```

## TLDR
@JoshuaKGoldberg, thanks for correcting. I misunderstood that the problem was that the doc didn't tell users where to find the plugins.
@bradzacher, sorry to tell you that I feel a little bit unwelcomed because I couldn't react with emoji. This project is great, I hope that we have more contributors who fill the issues.

### Reproduction Repository Link

https://github.com/PabloLION/test-repo/tree/node/eslint-typescript/plugins

### Repro Steps

# REPEATING [Issue Description](#Issue-Description)
1. Run `yarn init` to create a whatever new repo.
2. Install packages with `yarn` following [Getting Started - Linting your TypeScript Codebase](https://typescript-eslint.io/docs/linting/)

   ```bash
     yarn add --dev eslint typescript @typescript-eslint/parser @typescript-eslint/eslint-plugin
   ```

3. Add this `eslintConfig` to `package.json`, then run `yarn lint`. The command works.

   ```json
   "eslintConfig": {
       "env": {
         "node": true
       },
       "extends": [
         "eslint:recommended",
         "plugin:@typescript-eslint/recommended"
       ],
       "parser": "@typescript-eslint/parser",
       "plugins": [
         "@typescript-eslint"
       ],
       "root": true
     },
   ```

4. Add these two lines to `eslintConfig.extends`, then the command fails.

   ```json
   "plugin:@typescript-eslint/recommended-requiring-type-checking",
   "plugin:@typescript-eslint/strict"
   ```

5. I also tested with `.eslintrc.js` and `.eslintrc.cjs`. Both failed with same error.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     |5.25.0 |
| `@typescript-eslint/parser`            |5.25.0 |
| `@typescript-eslint/scope-manager`     | NONE |
| `@typescript-eslint/typescript-estree` | NONE |
| `@typescript-eslint/type-utils`        |  NONE  |
| `@typescript-eslint/utils`             |NONE |
| `TypeScript`                           | 4.6.4 |
| `ESLint`                               | 8.16.0 |
| `node`                                 |  18.2.0  |

**Comment by PabloLION** [comment]
There are 4 versions in the commits of this branch. One can switch swiftly in testing.

**Comment by PabloLION** [comment]
### One more thing to mention
There are 4 versions in the commits of this branch. One can switch swiftly in testing.e

**Comment by JoshuaKGoldberg** [comment]
```
~/repos/test-repo$ git status
HEAD detached at 885e4cc
nothing to commit, working tree clean
~/repos/test-repo$ yarn install; yarn lint
yarn install v1.22.18
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.18s.
yarn run v1.22.18
$ eslint . --ext .ts

Oops! Something went wrong! :(

ESLint: 8.16.0

Error: Error while loading rule '@typescript-eslint/dot-notation': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Occurred while linting /home/josh/repos/test-repo/hello.ts
    at getParserServices (/home/josh/repos/test-repo/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:23:15)
    at create (/home/josh/repos/test-repo/node_modules/@typescript-eslint/eslint-plugin/dist/rules/dot-notation.js:85:81)
    at Object.create (/home/josh/repos/test-repo/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:42:20)
    at createRuleListeners (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:922:21)
    at /home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1104:31
    at Array.forEach (<anonymous>)
    at runRules (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1041:34)
    at Linter._verifyWithoutProcessors (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1389:31)
    at Linter._verifyWithConfigArray (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1729:21)
    at Linter.verify (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1471:65)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by JoshuaKGoldberg** [comment]
Hmm, I get a different error when testing. The one I get is that your ESLint config is missing `parserOptions` as described in https://typescript-eslint.io/docs/linting/type-linting.

```
~/repos/test-repo$ git status
HEAD detached at 885e4cc
nothing to commit, working tree clean

~/repos/test-repo$ yarn install
yarn install v1.22.17
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.15s.

~/repos/test-repo$ yarn lint
yarn run v1.22.17
$ eslint . --ext .ts

Oops! Something went wrong! :(

ESLint: 8.16.0

Error: Error while loading rule '@typescript-eslint/dot-notation': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Occurred while linting /home/josh/repos/test-repo/hello.ts
    at getParserServices (/home/josh/repos/test-repo/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:23:15)
    at create (/home/josh/repos/test-repo/node_modules/@typescript-eslint/eslint-plugin/dist/rules/dot-notation.js:85:81)
    at Object.create (/home/josh/repos/test-repo/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:42:20)
    at createRuleListeners (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:922:21)
    at /home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1104:31
    at Array.forEach (<anonymous>)
    at runRules (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1041:34)
    at Linter._verifyWithoutProcessors (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1389:31)
    at Linter._verifyWithConfigArray (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1729:21)
    at Linter.verify (/home/josh/repos/test-repo/node_modules/eslint/lib/linter/linter.js:1471:65)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

When I add the missing `parserOption` config in and run `tsc --init`, everything passes:

```
~/repos/test-repo$ yarn lint
yarn run v1.22.17
$ eslint . --ext .ts
Done in 0.82s.
```

This happens both on Node 16 and 18.

...and aww. Don't worry, we'll leave you plenty of options to emoji react in this issue! 🥰 

[patch.diff.txt](https://github.com/typescript-eslint/typescript-eslint/files/8749692/patch.diff.txt)

**Comment by bradzacher** [comment]
> sorry to tell you that I feel a little bit unwelcomed because I couldn't react with emoji.

Sorry!
Some users are stubborn and ignore us, choosing to continue to comment instead of raising an issue. Locking "forces" the users to follow the rules to get help.
We mean no offence by locking the issue - it was just an abundance of caution against "bad actors".

**Comment by PabloLION** [comment]
Thanks for the explanation. Now I see your action was quite reasonable 😔😔

**Comment by PabloLION** [comment]
> When I add the missing `parserOption` config in and run `tsc --init`, everything passes:

Same for me. Adding a `"parserOptions": { "project": ["./tsconfig.json"] }` to the `.eslintrc.js` or to the `package.json` solves my problem. It seems a user mistake. Thanks for the answer! ❤️❤️

**Comment by PabloLION** [comment]
Related to #5030 but I can't reopen that one. I searched in [the configuration page](https://typescript-eslint.io/docs/linting/configs/#built-in-configurations) but no `parserOptions` was mentioned. Can we add it to the page?

**Comment by JoshuaKGoldberg** [comment]
@PabloLION please see the _Getting Started_ docs linked in the website homepage: https://typescript-eslint.io/docs/. Specifically, https://typescript-eslint.io/docs/linting/type-linting suggests the `parserOptions`.

Same as the last issue, please don't post on unrelated issues. I'll wait a bit before locking this one so you can emoji react if you want. 🥰

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg I think we should have a dedicated page for configuring the parser. WDYT about another documentation request for that? I always have to go to the npm package's README.

**Comment by PabloLION** [comment]
> @PabloLION please see the _Getting Started_ docs linked in the website homepage: https://typescript-eslint.io/docs/. Specifically, https://typescript-eslint.io/docs/linting/type-linting suggests the `parserOptions`.

I saw both pages before. In the first page no "parserOptions" was mentioned and the second one TBH I didn't read when I was configuration.

From the perspective of an old user who tries to configure the linter for the first time, it's quite hard to find the relation between the two pages(I could already "Lint my TypeScript Codebase" and I want to configure something more. I saw "configuration" page and I started reading this page) and the error message was not very specific so that I thought it was a bug at first.

Please consider at least linking the "Linting your TypeScript Codebase" page in the "Configuration" page, but I'll respect you call whatever it is.

**Comment by PabloLION** [comment]
Imagining 1000 more programmers wants to configure the ESLint, I believe quite some of them would get lost like me and they'd have to ask about the error message pf some not installed package. The trouble caused by this outbalances the benefit from reading one or two less lines. This was my reason to add a link or mention the "parserOptions".

**Comment by JoshuaKGoldberg** [comment]
> quite hard to find the relation between the two pages

You're absolutely right. The two pages are a lot of text to read through and you're certainly not the only person bit by the messy documentation.

Again, though **please don't post unrelated questions on previously resolved issues**. There are several issues with these unrelated-thread-conversations:

* They are not very discoverable for other users researching the same problem
* They make it harder for us, the maintainers, to keep track of requests
* Nothing enforces you using the [issue templates](https://github.com/typescript-eslint/typescript-eslint/issues/new/choose) we put in place -- but those are necessary to make sure your reports include all necessary information

That's why we keep locking issues (and I just filed #5089). You're suggesting good things that we absolutely should take action on. You're just asking in the wrong place 🙂.

So, please, file a new issue on https://github.com/typescript-eslint/typescript-eslint/issues/new/choose. We'll be happy to engage with you there.
